src/components/ebay-pagination/test/test.browser.js

```diff
@@ -0,0 +1,110 @@
+        });
+
+        test('then it emits the marko event called pagination-previous', () => {
+            expect(spy.calledOnce).to.equal(true);
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el']);
+        });
+    });
+
+    describe('when the next button is clicked', () => {
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-next', spy);
+            testUtils.triggerEvent(nextButton, 'click');
+        });
+
+        test('then it emits the marko event called pagination-next', () => {
+            expect(spy.calledOnce).to.equal(true);
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el']);
+        });
+    });
+
+    describe('when the page number is clicked', () => {
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-select', spy);
+            testUtils.triggerEvent(pageItem, 'click');
+        });
+
+        test('then it emits the marko event called pagination-select', () => {
+            expect(spy.calledOnce).to.equal(true);
+            const eventData = spy.getCall(0).args[0];
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el', 'value']);
+            expect(eventData.value).to.be.equal('1');
+        });
+    });
+
+    describe('when the previous button is activated through keydown', () => {
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-previous', spy);
+            testUtils.triggerEvent(previousButton, 'keydown', 32);
+        });
+
+        test('then it emits the marko event called pagination-previous', () => {
+            expect(spy.calledOnce).to.equal(true);
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el']);
+        });
+    });
+
+    describe('when the next button is activated through keydown', () => {
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-next', spy);
+            testUtils.triggerEvent(nextButton, 'keydown', 32);
+        });
+
+        test('then it emits the marko event called pagination-next', () => {
+            expect(spy.calledOnce).to.equal(true);
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el']);
+        });
+    });
+
+    describe('when the page number is activated through keydown', () => {
+        let spy;
+        beforeEach(() => {
+            spy = sinon.spy();
+            widget.on('pagination-select', spy);
+            testUtils.triggerEvent(pageItem, 'keydown', 32);
+        });
+
+        test('then it emits the marko event called pagination-select', () => {
+            expect(spy.calledOnce).to.equal(true);
+            const eventData = spy.getCall(0).args[0];
+            expect(Object.keys(spy.args[0][0])).to.deep.equal(['el', 'value']);
+            expect(eventData.value).to.be.equal('1');
+        });
+    });
+});
```

Comment:
Run `yarn tr` locally to see the coverage report, and fill in the missing gaps where possible.